docs/about/overview.md

```diff
@@ -0,0 +1,50 @@
+# What is the MyST stack?
```

Comment:
Also, importantly, talks about the relationship between MyST and Jupyter Book. Worth mentioning in the title or first paragraph?

Reply:
We have a little of that content [here](https://mystmd.org/guide/background#how-do-jupyter-book-and-mystmd-relate) too, so we might be fine not to address it in this title.